src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,70 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $calendarGap;
+  padding: $calendarMainPadding;
+  padding-top: 25px;
+
+  max-width: calendarSize($squareSize, $calendarMainPadding, $calendarGap);
+
+  &__date {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    width: $squareSize;
+    height: $squareSize;
+
+    cursor: pointer;
+    background-color: $dateColor;
+    border: 1px solid $dateBorderColor;
+
+    transition: background-color  0.5s, transform 0.5s;
+
+    &:hover {
+      background-color: pink;
+
+      transform: translateY(-20px);
+    }
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i})::before {
+        content: "#{$i}";
+        font-family: $calendarFont;
+        font-size: 30px;
+      };
+    }
+  }
+
+  $weekdays:
+    "mon" 0,
+    "tue"
+    calendarSize($squareSize, $calendarMainPadding, $calendarGap)
+    - 6 * $squareSize - 2 * $calendarMainPadding - 5 * $calendarGap,
+    "wed"
```

Comment:
move this calculation in the cycle where you set needed properties for each modifier
For now this code looks too complicated